_studio/mfx_lib/scheduler/include/mfx_scheduler_core_thread.h

```diff
@@ -31,25 +34,56 @@ class mfxSchedulerCore;
 
 struct MFX_SCHEDULER_THREAD_CONTEXT
 {
-    MFX_SCHEDULER_THREAD_CONTEXT()
+    typedef std::function<void(MFX_SCHEDULER_THREAD_CONTEXT*)> SchedulerThreadFunc;
+
+    MFX_SCHEDULER_THREAD_CONTEXT() = delete;
+
+    MFX_SCHEDULER_THREAD_CONTEXT(mfxU32 id)
       : state(State::Waiting)
-      , pSchedulerCore(NULL)
-      , threadNum(0)
-      , threadHandle()
+      , threadNum(id)
+      , threadHandle([this]() {
+          start.get_future().wait();
+          if (threadFunc) threadFunc(this);
+      })
       , workTime(0)
       , sleepTime(0)
-    {}
+    {
+    }
+
+    void Start(SchedulerThreadFunc func)
+    {
+        threadFunc = std::move(func);
+        start.set_value();
+    }
+
+    void Stop(std::lock_guard<std::mutex>& /*lock*/)
+    {
+        state = MFX_SCHEDULER_THREAD_CONTEXT::Stopping;
+        taskAdded.notify_one();
+    }
+
+    ~MFX_SCHEDULER_THREAD_CONTEXT()
```

Comment:
let's follow rule of five and declare explicitly following:
- copy ctor
- move ctor
- copy =
- move =

Reply:
done